script/index.js

```diff
@@ -0,0 +1,152 @@
+
+    const NULL_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+    const res = await requestTracker.methods.query(rfAddr, GT_HEX, currentBlockNum).call()
+    // log(web3.utils.isAddress(res))
+    // log(res)
+
+    if (!web3.utils.isAddress(res)) throw new Error('Did not receive a valid Ethereum address from requestTracker.')
+    if (res === NULL_ADDRESS) throw new Error('No upcoming transactions registered in requestTracker!')
+
+    /// Validation
+    const isKnown = await requestFactory.methods.isKnownRequest(res).call()
+
+    let txRequest = new web3.eth.Contract(
+        TransactionRequestABI,
+        res
+    )
+    // const requestData = await txRequest.methods.requestData().call()
+    const requestData = await RequestData.from(txRequest)
+    // log(requestData.schedule.windowStart)
+
+    if (requestData.schedule.windowStart < await web3.eth.getBlockNumber()) {
+        throw new Error('Cannot monitor! Window is already started.')
+    }
+
+    if (requestData.schedule.windowStart >= (await web3.eth.getBlockNumber()) + maxBlocks) {
+        throw new Error('Too far away. Please change maxBlocks variable.')
+    }
+
+    if (requestData.meta.isCancelled) throw new Error('This transaction is cancelled.')
+
+    cache.set(res, requestData.schedule.windowStart)
+    // cache.get(res)
+
+    const running = true
+    while (running) {
+        var nextAddr = await requestTracker.methods.getNextRequest(requestFactory.options.address, txRequest.options.address).call() 
+        // log(nextAddr)
+
+        if (nextAddr === NULL_ADDRESS) {
+            break 
+        }
+
+        txRequest.options.address = nextAddr        
+
+        const rData = await RequestData.from(txRequest)
+
+        if (cache.get(nextAddr) === null) {
+            cache.set(nextAddr, rData.schedule.windowStart)
+        }
+    }
+
+    setInterval(() => {
+        const memory = cache.stored()
+        memory.forEach(async (addr) => {
+            const txR = new web3.eth.Contract(
+                TransactionRequestABI,
+                addr
+            )
+            const rData = await RequestData.from(txR)
+            if (rData.meta.isCancelled) {
+                cache.delete(addr)
+            }
+            if ((await web3.eth.getBlockNumber()) > rData.schedule.windowStart + rData.schedule.windowSize) {
+                cache.delete(addr)
+            }
+            if (rData.meta.wasCalled) {
+                cache.delete(addr)
+            }
+            if (rData.claimData.isClaimed && (await web3.eth.getBlockNumber()) <= rData.schedule.windowStart +rData.schedule.claimWindowSize) {
+                /// nothing
+                log('shouldnt hit here yet') 
+            }
+            if (rData.claimData.isClaimed && (await web3.eth.getBlockNumber()) > rData.schedule.windowStart +rData.schedule.claimWindowSize) {
+                log('executing...')
+                txR.methods.execute().send({
```

Comment:
special care should be taken to not send the `execute()` transaction if we've already sent one and it just hasn't been mined.